components/switch/script.js

```diff
@@ -0,0 +1,110 @@
+import components from 'coherent-gameface-components';
+import textInside from './templates/text-inside-template.html';
+import textOutside from './templates/text-outside-template.html';
+
+/**
+ * @class Switch
+ * Switch component, that allows you to switch between true and false
+ */
+class Switch extends HTMLElement {
+    constructor() {
+        super();
+
+        this._onClick = this._onClick.bind(this);
+    }
+
+    /**
+     * Called when the element is attached to the DOM
+     */
+    connectedCallback() {
+        this._disabled = this.hasAttribute('disabled');
+        this._checked = this.hasAttribute('checked');
+
+        //The type of the switch. There are currently 3 possible - default, inset and text-inside
+        this._type = this.getAttribute('type');
+
+        //Helpers for easy readability
+        this._isDefault = this._type !== 'inset' && this._type !== 'text-inside';
+        this._isInset = this._type === 'inset';
+        this._isTextInside = this._type === 'text-inside';
+
+        //Set the template based if the text is inside or outside
+        this.template = this._isTextInside ? textInside : textOutside;
+
+        ///Load the template
+        components
+            .loadResource(this)
+            .then((result) => {
+                this.template = result.template;
+                //Render the template
+                components.renderOnce(this);
+
+                //Set the elements of the switch we'll be changing depending if it's checked or not
+                this._switch = this.querySelector('.switch-toggle');
+                this._handle = this.querySelector('.switch-toggle-handle');
+                this._textChecked = this.querySelector('.switch-toggle-true');
+                this._textUnchecked = this.querySelector('.switch-toggle-false');
+
+                this._setup();
+            })
+            .catch((err) => console.error(err));
+    }
+
+    disconnectedCallback() {
+        this._removeEventListeners();
+    }
+
+    /**
+     * Sets up the switch based on the attributes it has and attaches the event listeners
+     */
+    _setup() {
+        if (this._isDefault) this._handle.classList.add('switch-toggle-handle-shadow');
+
+        if (this._isInset) this._switch.classList.add('switch-toggle-inset');
+
+        if (this._checked) this._toggleClasses();
+
+        if (this._disabled) this.toggleDisabled();
+
+        this._attachEventListeners();
+    }
+
+    _attachEventListeners() {
+        this.firstChild.addEventListener('mousedown', this._onClick);
+    }
+
+    _removeEventListeners() {
+        this.firstChild.removeEventListener('mousedown', this._onClick);
+    }
+
+    /**
+     * Changes the styles of the switch and if it's checked
+     */
+    _onClick() {
+        this._checked = !this._checked;
+        this.dispatchEvent(new CustomEvent('switch_toggle', { detail: this._checked }));
```

Comment:
Why do we need this event? We haven't documented it and we only use it in the tests to check if it works.
Maybe document it so that people can use it? Or remove it if you think that we don't really need it?

Reply:
The main difference between a checkbox and a switch is that checking a switch should follow an immediate action. For example if we have a switch to toggle between a light and dark theme, clicking the switch will change it immediately whereas after clicking the checkbox it's still expected to press a button to submit it. This is why the event, so we can listen for it and follow up with an action. 

Added a paragraph to the README.